spring-boot-project/spring-boot-properties-migrator/src/test/java/org/springframework/boot/context/properties/migrator/PropertiesMigrationReporterTests.java

```diff
@@ -144,6 +144,31 @@ void invalidReplacementHandled() throws IOException {
 		assertThat(report).doesNotContain("null");
 	}
 
+	@Test
+	void detectDeprecatedPropertyInMap() {
+		MutablePropertySources propertySources = this.environment.getPropertySources();
+		Map<String, Object> content = new LinkedHashMap<>();
+		content.put("status.http-mapping.down", 500);
+		content.put("status.http-mapping.out-of-service", 503);
+		content.put("status.http-mapping.warning", 500);
+		propertySources.addFirst(new MapPropertySource("status.http-mapping", content));
```

Comment:
If I change the name of the property source to be something else than this, the test fails. The `PropertySource` implementation and its name are irrelevant for this, it could be anything.

Reply:
Nothing is name specific, but yes it is specific to MapPropertySource

Reply:
It is. Replace `new MapPropertySource("status.http-mapping", content)` by `new MapPropertySource("test", content)` and you'll see. The name of the property source has nothing to do whatsoever with the prefix used by the keys.